### PR TITLE
MDK CMSIS RTOSv2

### PR DESCRIPTION
**Pull request by kojo1**
Add CMSIS RTOSv2 support
update net3 to net4 to support RTOSv2
Some clean ups

### CODE REVIEW

File: wolfssl/internal.h [hunk 1]
```diff
@@ -146,14 +146,14 @@
         /* do nothing */
 #elif defined(WOLFSSL_CMSIS_RTOS)
     #include "cmsis_os.h"
+#elif defined(WOLFSSL_CMSIS_RTOSv2)
+    #include "cmsis_os2.h"
+#elif defined(WOLFSSL_CMSIS_RTOSv2)
```
Comment (cconlon) [inline wolfssl/internal.h]: There are two identical WOLFSSL_CMSIS_RTOSv2 blocks here.  Please remove one.

File: wolfssl/internal.h [hunk 2]
```diff
@@ -146,14 +146,14 @@
...
+    #include "cmsis_os2.h"
 #elif defined(WOLFSSL_MDK_ARM)
-    #if defined(WOLFSSL_MDK5)
```
Comment (cconlon) [inline wolfssl/internal.h]: Will removing this include for "cmsis_os.h" break existing users builds?  In other words, were previous users defining WOLFSSL_MDK_ARM and WOLFSSL_MDK5 without defining WOLFSSL_CMSIS_RTOS?

File: wolfssl/test.h [hunk 3]
```diff
@@ -52,18 +52,19 @@
     #include <string.h>
     #include "rl_net.h"
     #define SOCKET_T int
-        typedef int socklen_t ;
-        static unsigned long inet_addr(const char *cp)
+    typedef int socklen_t ;
+    #define inet_addr wolfSSL_inet_addr
```
Comment (cconlon) [inline wolfssl/test.h]: Why was this define introduced (mapping inet_addr to wolfSSL_inet_addr)?  I don't see that it ever gets defined to anything else.

File: wolfssl/wolfio.h [hunk 4]
```diff
@@ -69,11 +69,12 @@
     #elif defined(FREESCALE_KSDK_MQX)
         #include <rtcs.h>
     #elif (defined(WOLFSSL_MDK_ARM) || defined(WOLFSSL_KEIL_TCP_NET))
-        #include "cmsis_os.h"
```
Comment (cconlon) [inline wolfssl/wolfio.h]: Will removing this break existing users builds?

File: wolfssl/test.h (hunk 3 above)
Comment (kojo1) [reply wolfssl/test.h]: MDK has inet_addr, but has mismatched not cast-able return value type. 
wolfSSL_inet_addr is for avoiding the conflict,  providing functionally 
compatible to standard inet_addr.

File: wolfssl/internal.h (hunk 1 above)
Comment (kojo1) [reply wolfssl/internal.h]: Removed the duplication.

File: wolfssl/test.h (hunk 3 above)
Comment (cconlon) [reply wolfssl/test.h]: Ok, thanks

File: wolfssl/internal.h (hunk 1 above)
Comment (cconlon) [reply wolfssl/internal.h]: Thanks

File: wolfssl/internal.h (hunk 2 above)
Comment (kojo1) [reply wolfssl/internal.h]: Rollbacked

File: wolfssl/wolfio.h (hunk 4 above)
Comment (kojo1) [reply wolfssl/wolfio.h]: Rollbacked

File: wolfcrypt/benchmark/benchmark.c [hunk 5]
```diff
@@ -3989,15 +3989,32 @@ static void bench_rsa_helper(int doAsync, RsaKey rsaKey[BENCH_MAX_PENDING],
 #ifndef WOLFSSL_RSA_VERIFY_ONLY
     DECLARE_VAR_INIT(message, byte, len, messageStr, HEAP_HINT);
 #endif
-
-    DECLARE_ARRAY_DYNAMIC(enc, byte, BENCH_MAX_PENDING, rsaKeySz, HEAP_HINT);
+    #if !defined(WOLFSSL_MDK5_COMPLv5)
+	  /* MDK5 compiler regard this as a executable statement, and does not allow declarations after the line. */
+        DECLARE_ARRAY_DYNAMIC_DEC(enc, byte, BENCH_MAX_PENDING, rsaKeySz, HEAP_HINT);
+	  #else
```
Comment (cconlon) [inline wolfcrypt/benchmark/benchmark.c]: It looks like there may be some hard tabs around this area.  Could you convert them to 4 spaces to help with readability?  Thanks!

File: wolfcrypt/benchmark/benchmark.c [hunk 6]
```diff
@@ -3989,15 +3989,32 @@ static void bench_rsa_helper(int doAsync, RsaKey rsaKey[BENCH_MAX_PENDING],
...
+	      byte* enc[BENCH_MAX_PENDING];
+	      int idxenc;
+	  #endif
     #if !defined(WOLFSSL_RSA_VERIFY_INLINE) && \
                     !defined(WOLFSSL_RSA_PUBLIC_ONLY)
-        DECLARE_ARRAY_DYNAMIC(out, byte, BENCH_MAX_PENDING, rsaKeySz, HEAP_HINT);
+        #if !defined(WOLFSSL_MDK5_COMPLv5) 
+	      /* MDK5 compiler regard this as a executable statement, and does not allow declarations after the line. */
+            DECLARE_ARRAY_DYNAMIC_DEC(out, byte, BENCH_MAX_PENDING, rsaKeySz, HEAP_HINT);
+		    #else
```
Comment (cconlon) [inline wolfcrypt/benchmark/benchmark.c]: Please fix hard tabs here as well, thanks.

File: wolfcrypt/benchmark/benchmark.c (hunk 5 above)
Comment (kojo1) [reply wolfcrypt/benchmark/benchmark.c]: removed

File: wolfcrypt/benchmark/benchmark.c (hunk 6 above)
Comment (kojo1) [reply wolfcrypt/benchmark/benchmark.c]: removed

**Pull request merged**